### PR TITLE
test: add a few smoke tests for profiling

### PR DESCRIPTION
**Pull request by seemk**
Verifies the types received from native extension and that profiling data is exported.

The logs exporter is not tested as we're about to throw it away anyway in favor of pprof.

### CODE REVIEW

**Review by Rauno56 — APPROVED** [review]
Please address the comments and feel free to request another review if you think is necessary.
Will approve now because profiling API is not stable. We def need to address the API differences with tracing before going stable - one of them is introduced in this PR(exporters configuration option), the other(start-stop) was there before.

File: test/profiling/extension.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    assert(stacktraces.length >= 10);
+
+    for (const { stacktrace, timestamp } of stacktraces) {
+      // Don't bother checking for span and trace ID here.
+      assert.equal(typeof timestamp, 'string');
+      assert.equal(typeof stacktrace, 'string');
+
+      const lines = stacktrace.split('\n');
+      assert.deepStrictEqual(lines[0], '');
```
[the remark this answers is not in the document]
Comment (seemk) [reply test/profiling/extension.test.ts:67]: [Java specifics](https://github.com/signalfx/gdi-specification/blob/main/specification/semantic_conventions.md#call-stack-format-for-text-data-format) mostly. We add these lines on our own currently to comply with the spec/backend.

File: test/profiling/profiling.test.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      let sendCallCount = 0;
+      const exporter: ProfilingExporter = {
+        send(profilingData: ProfilingData) {
+          const { stacktraces } = profilingData;
+          stacktracesReceived += stacktraces.length;
+          sendCallCount += 1;
+        },
+      };
+
+      const { stop } = startProfiling({
```
[the remark this answers is not in the document]
Comment (seemk) [reply test/profiling/profiling.test.ts:90]: I'll do breaking changes in a separate PR

File: test/profiling/extension.test.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    while (durationNanos < 10 * 1e6) {
+      const begin = hrtime.bigint();
+      spin();
+      durationNanos += Number(hrtime.bigint() - begin);
+    }
+
+    const result = extension.collect();
+    // The types might not be what is declared in typescript, a sanity check.
+    assert.equal(typeof result, 'object');
+    const { stacktraces, startTimeNanos } = result;
```
[the remark this answers is not in the document]
Comment (seemk) [reply test/profiling/extension.test.ts:53]: Added a test for it

File: src/profiling/types.ts [hunk 4]
```diff
@@ -51,6 +51,7 @@ export interface ProfilingOptions {
   collectionDuration: number;
   debugExport: boolean;
   resource: Resource;
+  exporters?: ProfilingExporter[];
```
[the remark this answers is not in the document]
Comment (seemk) [reply src/profiling/types.ts]: Good point, replaced it with a factory

File: test/profiling/extension.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Rauno56) [reply test/profiling/extension.test.ts:70]: I meant perhaps you could mention that in a comment in the code as well.

File: src/profiling/types.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Rauno56) [reply src/profiling/types.ts]: This is still different from tracing - Exporter vs list of them.
I see you want to trivially support adding the `DebugExporter`, hence the list. Could you make it also accept a single exporter?

[the remark this answers is not in the document]
Comment (seemk) [reply src/profiling/types.ts]: I think tracing API has it backwards, we should add support for a list of processors there - a backwards compatible change.

Metrics will have an array as well: https://github.com/signalfx/splunk-otel-js/blob/metrics-next/src/metrics/index.ts#L37

File: test/profiling/extension.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (seemk) [reply test/profiling/extension.test.ts:72]: Added a comment, but this will be gone anyway once pprof is out

**Pull request merged**